internal/worker/server.go

```diff
@@ -546,7 +542,9 @@ func (s *Server) FinishJob(token uuid.UUID, result json.RawMessage) error {
 		logrus.Errorf("error finding job status: %v", err)
 	} else {
 		statusCode := clienterrors.GetStatusCode(jobResult.JobError)
```

Comment:
I just realized that by using `jobStatus()` in `FinishJob()` we are potentially missing various errors set in `jobResult.JobError` by the job-type-specific status methods, such as `OSBuildJobStatus`.

What would probably make sense is to have a `switch` block here, based on the job type determined using `JobType()` (which will eliminate the need for `getMetricsJobType()`). And subsequently call the appropriate job-type-specific status method (which will eliminate the need for `getBuildJobArch`) to get the job status and result.

It will be one more call to the DB compared to the current state, but that may be acceptable.

Reply:
Oh yeah this is a good point, I'm going to have to rework each of the `JobStatus` functions to return the channel since we need that, but that's not a big deal.